src/components/analyses/ShareWithSupportDialog.js

```diff
@@ -0,0 +1,577 @@
+                            )}
+                    </React.Fragment>
+                )}
+
+                {shareWithSupport && (
+                    <React.Fragment>
+                        <AnalysisInfo
+                            analysis={analysis}
+                            name={name}
+                            email={email}
+                        />
+                        <TextField
+                            id={build(baseDebugID, ids.COMMENTS)}
+                            placeholder={t("comments")}
+                            style={{ margin: 8 }}
+                            fullWidth
+                            multiline
+                            margin="normal"
+                            variant="outlined"
+                            onChange={(event) => {
+                                setComment(event.target.value);
+                            }}
+                            InputLabelProps={{
+                                shrink: true,
+                            }}
+                        />
+                        <FormControlLabel
+                            control={
+                                <Checkbox
+                                    checked={enableSubmit}
+                                    onChange={(event) =>
+                                        setEnableSubmit(event.target.checked)
+                                    }
+                                    value={enableSubmit}
+                                />
+                            }
+                            label={<ShareDisclaimer />}
+                        />
+                    </React.Fragment>
+                )}
+            </DialogContent>
+            <DialogActions>
+                <Button
+                    variant="outlined"
+                    disabled={loading}
+                    onClick={() => {
+                        setShareWithSupport(false);
+                        onClose();
+                    }}
+                >
+                    {t("cancel")}
+                </Button>
+
+                {shareWithSupport ? (
+                    <Button
+                        variant="contained"
+                        color="primary"
+                        disabled={!enableSubmit || loading}
+                        onClick={() => {
+                            onShareWithSupport(analysis, comment);
+                        }}
+                    >
+                        {loading ? <CircularProgress size={25} /> : t("submit")}
+                    </Button>
+                ) : (
+                    <Button
+                        variant="contained"
+                        color="primary"
+                        onClick={() => {
+                            setShareWithSupport(true);
+                        }}
+                    >
+                        {t("needHelp")}
+                    </Button>
```

Comment:
Since this is mostly static content in a dialog, I'm not sure how much we really want to test this, but in the very least it would be helpful to have static IDs on these dialog action buttons if we want to make sure the dialog at least can open, shows something, and then close it.